src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -922,11 +922,18 @@ void ConnectionHandlerImpl::OnConnectionEnded(
          session_map.end() != session_it; ++session_it) {
       const uint32_t session_key = KeyFromPair(connection_id, session_it->first);
       const ServiceList &service_list = session_it->second.service_list;
-      for (ServiceList::const_iterator service_it = service_list.begin(), end =
+#ifdef OS_WINCE
+      for (int i = service_list.size() - 1; i >= 0; i--) {
```

Comment:
@smartdevice475 @luwanjia 
Why not use `cont_iterator` here? such as:
```  
for (ServiceList::const_iterator service_it = service_list.end(),begin =
    service_list.begin(); service_it != begin; --service_it) {
````
Dose WinCE port not support `iterator`?